src/local-history/local-history-manager.ts

```diff
@@ -78,43 +76,43 @@ export class LocalHistoryManager {
     /**
      * View all history of the active editor.
      */
-    public viewAllForActiveEditor(textEditor: vscode.TextEditor): void {
+    public viewAllHistoryForActiveEditor(textEditor: vscode.TextEditor): void {
 
         if (vscode.workspace.getWorkspaceFolder(textEditor.document.uri) !== undefined) {
-            const workspaceFolderPath: string = vscode.workspace.getWorkspaceFolder(textEditor.document.uri)!.uri.fsPath;
-            const activeEditor = path.parse(textEditor.document.fileName);
+            const hashedEditorPath = checksum(textEditor.document.fileName);
+            const hashedFolderPath = path.join(os.homedir(), '.local-history', hashedEditorPath);
 
-            let items: vscode.QuickPickItem[] = this.historyFiles.
-                filter(item => item.parentFileName === (activeEditor.base)).
-                map(item => ({
+            this.loadHistoryForActiveEditor(hashedFolderPath).then(() => {
+                let items: vscode.QuickPickItem[] = this.historyFilesForActiveEditor.map(item => ({
                     label: `$(calendar) ${item.timestamp}`,
-                    description: vscode.workspace.asRelativePath(item.uri)
+                    description: path.basename(item.uri)
                 }));
 
-            if (items.length === 0) {
-                vscode.window.showInformationMessage(`No local history file found for '${activeEditor.base}'`);
-                return;
-            }
+                if (items.length === 0) {
+                    vscode.window.showInformationMessage(`No local history file found for '${path.basename(textEditor.document.fileName)}'`);
+                    return;
+                }
+
+                // Limits the number of entries based on the preference 'local-history.maxEntriesPerFile`.
+                items = items.slice(0, this.localHistoryPreferenceService.maxEntriesPerFile);
+                vscode.window.showQuickPick(items,
+                    {
+                        placeHolder: `Please select a local history revision for '${path.basename(textEditor.document.fileName)}'`,
+                        matchOnDescription: true,
+                        matchOnDetail: true
+                    }).then((selection) => {
+                        // User made final selection.
+                        if (!selection) {
+                            return;
+                        }
+
+                        // Get the file system path for the selection.
+                        const selectionFsPath = path.join(hashedFolderPath, selection.description!);
 
```

Comment:
These changes look unrelated? Is it perhaps due to formatting issues?

Reply:
Yes.